# Palmer Penguins
***
This notebook contains my analysis of the famous palmer penguins dataset.

![Palmer](https://upload.wikimedia.org/wikipedia/commons/thumb/2/24/Antarctica_relief_location_map.jpg/240px-Antarctica_relief_location_map.jpg)

The data set is avaialble [on Github](https://allisonhorst.github.io/palmerpenguins/).

The Palmer Penguin data set was collected by Dr. Kristen Gorman at Palmer Station in Antarctica. 
The data set was originally created with the intention of studying penguin foraging behaviour and environmental variability relationships.
This data set is now commonly used to study data analytics and machine learning.
The data set looks at the following variables:
|Variable      |Type/Unit                 |
|--------------|:------------------------ |
|Species       |Adelie/ Chinstrap/ Gentoo |
|Island        |Torgersen/ Biscoe/ Dream  |
|Bill length   | mm                       |
|Bill depth    | mm                       |   
|Flipper length| mm                       |
|Body mass     | g                        |
|Sex           | male/female/NaN          |

The three species of penguins are pictured below:
<div style="display: flex; justify-content: space-between;">
    <div style="text-align: center;">
        <strong>Adelie</strong>
        <br>
        <img src="https://upload.wikimedia.org/wikipedia/commons/8/82/Manchot_Ad%C3%A9lie_-_Adelie_Penguin.jpg" style="max-width: 100%; height: 200px;">
    </div>
    <div style="text-align: center;">
        <strong>Chinstrap</strong>
        <br>
        <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/cd/Chinstrap_Penguin.jpg/160px-Chinstrap_Penguin.jpg" style="max-width: 100%; height: 200px;">
    </div>
    <div style="text-align: center;">
        <strong>Gentoo</strong>
        <br>
        <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/0/00/Brown_Bluff-2016-Tabarin_Peninsula%E2%80%93Gentoo_penguin_%28Pygoscelis_papua%29_03.jpg/800px-Brown_Bluff-2016-Tabarin_Peninsula%E2%80%93Gentoo_penguin_%28Pygoscelis_papua%29_03.jpg"style="max-width: 100%; height: 200px;">
    </div>
</div>

The penguins had their species identified, were sexed, measured and the island on which they were found was recorded.
The entire data set is comprised of 344 penguins:
|Species | Amount|
|--------|:------:|
|Adelie  |152|
|Chinstrap|124|
|Gentoo  |68|

This notebook will give an overview of the data set, suggest vairiables for modelling and create approrprate bar and histogram charts. It will then provide further analysis of 2 appropriate variables to draw correlations and reasonable conclusions.

Note: Soruces are provided accordingly in some instances these were used in conjuction with AI to gain a simpler understanding of how to implement.
Sources:
Introduction to the data set https://cran.r-project.org/web/packages/palmerpenguins/readme/README.html
HTML Image formatting and insertion : https://www.w3schools.com/html/html_images.asp

## Imports
***
we use pandas for the dataframe data structure

it allows us to investigate CSV files amongst other features

In [1]:
# Data frames.
import pandas as pd

## Load Data
Load the palmer penguins data set from a URL.

In [2]:
# Load the penguins data set.
df = pd.read_csv("https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv")

The data is now loaded and we can inspect it.

In [3]:
# Lets have a look.
df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
339  Gentoo     Biscoe             NaN            NaN                NaN   
340  Gentoo     Biscoe            46.8           14.3              215.0   
341  Gentoo     Biscoe            50.4           15.7              222.0   
342  Gentoo     Biscoe            45.2           14.8              212.0   
343  Gentoo     Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  
0         3750.0    MALE  
1         3800.0  FEMALE  
2         3250.0  FEMALE  
3            NaN     NaN  
4         3450.0  FEMALE  
..           ...     ...  
339          NaN     NaN  
340       4850.0  FEMALE  
341       5750.0    MALE  
342       5200.0  FEMALE  
343       5400.0    MALE  

[344 rows x 7 columns]

## Inspect Data
***

In [4]:
#Look at the first row.
df.iloc[0]

species                 Adelie
island               Torgersen
bill_length_mm            39.1
bill_depth_mm             18.7
flipper_length_mm        181.0
body_mass_g             3750.0
sex                       MALE
Name: 0, dtype: object

In [9]:
# Let's undertand the types of data we have in this data set.
df.dtypes

species               object
island                object
bill_length_mm       float64
bill_depth_mm        float64
flipper_length_mm    float64
body_mass_g          float64
sex                   object
dtype: object

We can see above that the data is either in the format as floating point numbers or objects. Objects are non-heterogenous data and are common for strings. Floating point numbers should be treated with caution especially if further calculations or manipulations of the data are performed. This is because binary systems are commonly not accurate due to the 64 bit limits and therefore can lead to rounding.

Interestingly one of the variables that we do not always have a clear value for is sex. This is most likely to the surveyour not being able to sex the penguin at the time. This analysis would like to take the penguin measurements to:
1. Determine trends of penguin measurements between species.
2. Determine differences if penguin measurement between sexes.
3. Use these findings to give an estimated guess as to the missing sexes of the penguins.

Source: https://www.geeksforgeeks.org/floating-point-error-in-python/ 

In [5]:
# Sex of penguins 
df['sex']

0        MALE
1      FEMALE
2      FEMALE
3         NaN
4      FEMALE
        ...  
339       NaN
340    FEMALE
341      MALE
342    FEMALE
343      MALE
Name: sex, Length: 344, dtype: object

In [13]:
# Count the number of penguins of each sex  
Value_counts = df['sex'].value_counts()
# How many of these sexes are there in total?
Total_Identifiable_Penguins = Value_counts.sum()

print(Value_counts)

print(Total_Identifiable_Penguins)

sex
MALE      168
FEMALE    165
Name: count, dtype: int64
333


As one can see above there are 333 penguins that have been sexed and using the data here one would hope we can make a reasonable prediction as to the sex of the 11 NaN penguins.

In [7]:
# Descibe the data set.
df.describe()


bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

# this will be big

*you might have a paragaph pragraphs* have sentences.you might have a paragaph pragraphs have sentences.
you might have a paragaph pragraphs have sentences.
you might have a paragaph pragraphs have sentences.
you might have a paragaph pragraphs have sentences.
you might have a paragaph pragraphs have sentences.
you might have a paragaph pragraphs have sentences.
you might have a paragaph pragraphs have sentences.
you might have a paragaph pragraphs have sentences.

if you want a second paragraph leave a blank line! if you want a second paragraph leave a blank line!
if you want a second paragraph leave a blank line!
if you want a second paragraph leave a blank line!
if you want a second paragraph leave a blank line!
if you want a second paragraph leave a blank line!
if you want a second paragraph leave a blank line!if you want a second paragraph leave a blank line!
if you want a second paragraph leave a blank line!
**if you want a second paragraph leave a blank line!**



## this will be slightly smaller 
- #Bullet
- Point 
- Lists 

### this will be slightly smaller again
1. numbereed bullets
1. just use numbers.
1. this can be all sorted 1. 
1. how complex

> The definition of stupidity...... 
#### this will be slightly smaller again again
##### this will be slightly smaller again again again
###### this will be slightly smaller again again again again

## Tables

***

|Species   | Bill Length (mm) |bODY mass (g) |          
|----------|-----------------:|-------------:|
|Adelie    |             38.8 |          3701|
|Chinstrap |             48.8 |          3733|
|Gentoo    |             47.5 |          5076|

```  bash
$ ptyhon.py 
```

## Math 
***
$ f(x) = x^2$



***
###
End 